packages/babel-cli/src/_babel-node.js

```diff
@@ -48,6 +48,14 @@ let replPlugin = () => ({
       if (path.node.kind !== "var") {
         throw path.buildCodeFrameError("Only `var` variables are supported in the REPL");
       }
+    },
+
+    Directive(path) {
+      if (_.isUndefined(path.node.loc)) {
+        // If the executed code doesn't evaluate to a value,
+        // prevent implicit strict mode from printing 'use strict'.
+        path.parent.body.unshift(t.expressionStatement(t.identifier("undefined")));
```

Comment:
If we do this, it should be

    path.parentPath.pushContainer(t.expressionStatement(t.identifier("undefined"));

but I actually think we'd be better off reworking this to be

    Program(path){
        if (path.get('body.body').some((child) => child.isExpressionStatement()) return;

        path.pushContainer(t.expressionStatement(t.identifier("undefined"));
    }